pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -1420,17 +1553,17 @@ func (c *FlowContext) getSubnetKey(item *awsclient.Subnet) (zoneName, subnetKey
 			for _, key := range []string{IdentifierZoneSubnetWorkers, IdentifierZoneSubnetPublic, IdentifierZoneSubnetPrivate} {
 				switch key {
 				case IdentifierZoneSubnetWorkers:
-					if value == helper.GetSuffixSubnetWorkers() {
+					if value == fmt.Sprintf("%s-%s", c.namespace, helper.GetSuffixSubnetWorkers()) {
```

Comment:
can this value be changed freely ? What happens to existing clusters that have already a state and the key will change ?

Reply:
I'm wondering how this should have worked before. The tag is set [here](https://github.com/gardener/gardener-extension-provider-aws/blob/ef563c1cd252449234644121e9c301d4c60712a1/pkg/controller/infrastructure/infraflow/context.go#L315) and could have never matched before. Or am I missing something.